framework/src/materials/Material.C

```diff
@@ -171,8 +175,26 @@ Material::getOutputs()
 void
 Material::computeProperties()
 {
-  for (_qp = 0; _qp < _qrule->n_points(); ++_qp)
+  // If this Material has the _constant_on_elem flag set, we take the
```

Comment:
I think this duplicates the functionality of `computeProperties()` itself, i.e. if you need to set up something before the qp-loop, you are already supposed to override `computeProperties()`.  So from that standpoint I don't think it makes sense...